pkg/manager/runtime/handler.go

```diff
@@ -47,47 +57,61 @@ func (p *Server) CreateRuntime(ctx context.Context, req *pb.CreateRuntimeRequest
 	}
 
 	// get response
-	pbRuntime, err := p.getRuntimePbWithLabel(runtimeId)
+	runtime, err := p.getRuntimePbWithLabelAndCredential(runtimeId)
 	if err != nil {
 		logger.Errorf("CreateRuntime: %+v", err)
 		return nil, status.Errorf(codes.Internal, "CreateRuntime: %+v", err)
 	}
 	res := &pb.CreateRuntimeResponse{
-		Runtime: pbRuntime,
+		Runtime: runtime,
 	}
 	return res, nil
 }
 
 func (p *Server) DescribeRuntimes(ctx context.Context, req *pb.DescribeRuntimesRequest) (*pb.DescribeRuntimesResponse, error) {
 	// validate req
-	err := validateDescribeRuntimeRequest(req)
+	err := validateDescribeRuntimesRequest(req)
+	offset := utils.GetOffsetFromRequest(req)
+	limit := utils.GetLimitFromRequest(req)
 	if err != nil {
 		logger.Errorf("DescribeRuntimes: %+v", err)
 		return nil, status.Errorf(codes.InvalidArgument, "DescribeRuntimes: %+v", err)
 	}
-	var runtimeIds []string
-	// get runtime env ids by selector
-	if req.Selector != nil {
-		runtimeIds, err = p.getRuntimeIdsBySelectorString(req.Selector.GetValue())
-		if err != nil {
-			logger.Errorf("DescribeRuntimes: %+v", err)
-			return nil, status.Errorf(codes.Internal, "DescribeRuntimes: %+v", err)
-		}
+	selectorMap, err := SelectorStringToMap(req.Label.GetValue())
+	if err != nil {
+		logger.Errorf("DescribeRuntimes: %+v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "DescribeRuntimes: %+v", err)
 	}
-	// get runtime envs
-	pbRuntimes, count, err := p.getRuntimePbsWithoutLabelByReqAndId(req, runtimeIds)
+
+	var runtimes []*models.Runtime
+	var count uint32
+	query := p.Db.
+		Select(models.RuntimeColumnsWithTablePrefix...).
+		From(models.RuntimeTableName).
+		Offset(offset).
+		Limit(limit).
+		Where(manager.BuildFilterConditionsWithPrefix(req, models.RuntimeTableName))
+
+	query = db.AddJoinFilterWithMap(query, models.RuntimeTableName, models.RuntimeLabelTableName, RuntimeIdColumn,
+		models.ColumnLabelKey, models.ColumnLabelValue, selectorMap)
+
+	_, err = query.Load(&runtimes)
 	if err != nil {
 		logger.Errorf("DescribeRuntimes: %+v", err)
 		return nil, status.Errorf(codes.Internal, "DescribeRuntimes: %+v", err)
 	}
-	// get runtime envs label
-	pbRuntimes, err = p.getRuntimePbsLabel(pbRuntimes)
+
+	count, err = query.Count()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "DescribeRuntimes: %+v", err)
+	}
+	pbRuntime, err := p.getRuntimePbsWithLabelAndCredential(runtimes)
```

Comment:
`getRuntimePbsWithLabelAndCredential` change to `formatRuntimeSet` like `formatRepoSet `